.github/workflows/build_and_test_docker_image.yml

```diff
@@ -0,0 +1,125 @@
+
+on:
+  push:
+    branches: [ main ]
+
+env:
+  STAGING_REGISTRY: 264252446756.dkr.ecr.us-east-1.amazonaws.com
+  STAGING_REPOSITORY: credential-helper-staging
+  AWS_REGION: us-east-1
+
+permissions:
+    contents: read
+    packages: write
+    id-token: write   # This is required for requesting the JWT for AWS authentication
+
+jobs:
+  build-and-push:
+    name: Build Docker Images and Push to Staging Repository
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-24.04
+            platform: amd64
+          - os: ubuntu-24.04-arm
+            platform: arm64
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@master
+        with:
+          role-to-assume: arn:aws:iam::264252446756:role/GithubActionsAccessRole
+          aws-region: ${{ env.AWS_REGION }}
+          
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v2
+
+      - name: Extract Version from Makefile
+        id: extract-version
+        run: |
+          VERSION=$(grep '^VERSION=' Makefile | cut -d'=' -f2)
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+          echo "Extracted version: $VERSION"
+
+      - name: Build and Push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./docker_image_resources/Dockerfile
+          platforms: linux/${{ matrix.platform }}
+          push: true
+          build-args: |
+            VERSION=${{ steps.extract-version.outputs.version }}
+          tags: |
+            ${{ env.STAGING_REGISTRY }}/${{ env.STAGING_REPOSITORY }}:${{ github.sha }}-${{ matrix.platform }}
+          provenance: false # provenance must be disabled in order to prevent manifest creation failures
+
+  test-and-scan:
+    name: Test and Scan Docker Images
+    needs: build-and-push
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-24.04
+            platform: amd64
+          - os: ubuntu-24.04-arm
+            platform: arm64 
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@master
+        with:
+          role-to-assume: arn:aws:iam::264252446756:role/GithubActionsAccessRole
+          aws-region: ${{ env.AWS_REGION }}
+          
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v2
+      
+      - name: Set up Test Environment
+        env:
+          PCA_ARN: arn:aws:acm-pca:us-east-1:264252446756:certificate-authority/807cc589-2b5a-4976-a0d8-eeca654e7916 
+        run: |
+          docker pull ${{ env.STAGING_REGISTRY }}/${{ env.STAGING_REPOSITORY }}:${{ github.sha }}-${{matrix.platform}}
```

Comment:
Due to how ECR works with eventual consistency, you might need an `until` loop here.